src/services/workflow.service.js

```diff
@@ -29,6 +31,70 @@ class LiveWorkflowService extends GQuery {
   }
 }
 
-const workflowService = new LiveWorkflowService()
+class SubscriptionWorkflowService extends GQuery {
+  constructor () {
+    super(/* enableWebSockets */ true)
+    /**
+     * @type {object}
+     */
+    this.observable = null
+  }
+
+  destructor () {
+    if (this.observable) {
+      this.observable.unsubscribe()
+    }
+    this.observable = null
+  }
+
+  subscribe (view, query) {
+    this.observable = this.request()
+    return super.subscribe(view, query)
+  }
+
+  recompute () {
+    this.destructor()
+    super.recompute()
+    // query has been recomputed, so subscribe to request again
+    this.observable = this.request()
+  }
+
+  request () {
+    /**
+     * Perform a REST GraphQL request for all subscriptions.
+     */
+    if (process.env.NODE_ENV !== 'production') {
+      console.debug('graphql request:', this.query)
+    }
+    if (!this.query) {
+      return null
+    }
+    const vm = this
+    return this.apolloClient.subscribe({
+      query: this.query,
+      fetchPolicy: 'no-cache'
+    }).subscribe({
+      next (response) {
+        // commit results
+        store.dispatch(
+          'workflows/set',
+          response.data.workflows
+        )
+        // set all subscriptions to active
+        vm.subscriptions
+          .filter(s => s.active === false)
+          .forEach(s => { s.active = true })
+        // run callback functions on the views
+        vm.callbackActive()
+      },
+      error (err) {
+        store.dispatch(
+          'setAlert',
+          new Alert(err.message, null, 'error')
+        )
+      }
+    })
+  }
+}
 
-export { workflowService }
+export { LiveWorkflowService, SubscriptionWorkflowService }
```

Comment:
We were exporting a singleton, that once imported would exist for the whole app.

Now we export classes, let users decide whatever they want to do with these (mock, test, extend, instantiate, create a singleton, etc).

Vue has a way to create application globals via `Vue.prototype.$applicationSingleton`, which has effectively the same result as exporting a singleton, with more flexibility if necessary.

Plus, it will be available to the whole application as `this.$workflowService`. The `$` is per convention.

ps: must be used with care this global objects in Vue, as they docs state

Reply:
Can we get rid of the `LiveWorkflowService` now?

Reply:
I can't recall why I left. Not sure if it was still being used for testing/mocking, or if I was scared that the WebSockets could fail and we would have to revert to the old service :smile: will take another look to confirm.